text/0000-deprecation-ember-string.md

```diff
@@ -0,0 +1,62 @@
+- Start Date: 2017-07-14
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+This PR proposes the deprecation of `Ember.String`. The following methods will be moved to an addon:
+
+- `camelize`
+- `capitalize`
+- `classify`
+- `dasherize`
+- `decamelize`
+- `fmt`
+- `loc`
+- `underscore`
+- `w`
+
+The remaining methods (`htmlSafe` and `isHTMLSafe`) will be moved to `@ember/component`.
+
+In both cases, the String prototype won't be extended anymore.
```

Comment:
> I don't think the point of a codmod is really to eliminate all human review of code 

@bryanaka Fully agree! But I think it should be quite conservative, in the sense that it may not convert everything, but for sure it won't do any harm, like making plain wrong transformations. Imagine a not so great test coverage, and wrong transformations are not detected, leading to real bugs...

I do think a codemod could add some real benefits when converting imports (new Ember Module API), or destructurings (`const { String: { dasherize } } = Ember;`) because they are easy to analyze statically. (See https://github.com/cibernox/ember-k-codemod). But I would question that a codemod would work well detecting the String prototype extensions, possibly doing more harm than adding value, as probably nobody writes `'foo bar'.dasherize()`, so all use cases are comparable to the example given above (dasherize or any other method called on a variable rather than on a string literal) that you cannot reliably convert.

So **tl;dr** yes, the prototype extensions can only be removed for Ember 3.0, and while we are on 2.x the deprecation notices should be used to refactor our code, where possible with a codemod, but I think also to a quite large extend manually.